sdk/keyvault/Azure.Security.KeyVault.Secrets/Readme.md

```diff
@@ -0,0 +1,254 @@
+```
+> SystemCredential():
+> Uses the environment variables previosly set (`AZURE_CLIENT_ID`, `AZURE_CLIENT_SECRET`, and `AZURE_TENANT_ID`)
+
+## Key concepts
+### Secret
+A secret is the fundamental resource within Azure KeyVault. From a developer's perspective, Key Vault APIs accept and return secret values as strings.
+
+### Secret Client:
+An asynchronous and synchronous SecretClient client exists in the SDK allowing for selection of a client based on an application's use case. Once you've initialized a SecretClient, you can interact with the primary resource types in Key Vault.
+
+## Examples
+The following section provides several code snippets using the above created `client`, covering some of the most common Azure Key Vault Secret service related tasks, including:
+
+* [Create a Secret](#create-a-secret)
+* [Retrieve a Secret](#retrieve-a-secret)
+* [Update an existing Secret](#update-an-existing-secret)
+* [Delete a Secret](#delete-a-secret)
+* [List Secrets](#list-secrets)
+* [Async create a Secret](#async-create-a-secret)
+
+### Create a Secret
+`Set` creates a Secret to be stored in the Azure Key Vault. If a secret with the same name already exists, then a new version of the secret is created.
+
+```c#
+Secret secret = client.Set("secret-name", "secret-value");
+
+Console.WriteLine(secret.Name);
+Console.WriteLine(secret.Value);
+Console.WriteLine(secret.Version);
+Console.WriteLine(secret.Enabled);
+```
+
+### Retrieve a Secret
+`Get` retrieves a secret previously stored in the Key Vault.
+
+```c#
+Secret secret = client.Get("secret-name");
+
+Console.WriteLine(secret.Name);
+Console.WriteLine(secret.Value);
+```
+
+### Update an existing Secret
+`Update` updates a secret previously stored in the Key Vault.
+
+```c#
+Secret secret = new Secret("secret-name");
+
+// Clients may specify the content type of a secret to assist in interpreting the secret data when it's retrieved
+secret.ContentType = "text/plain";
+
+// You can specify additional application-specific metadata in the form of tags.
+secret.Tags["foo"] = "updated tag";
+
+SecretBase updatedSecret = client.Update(secret);
+
+Console.WriteLine(updatedSecret.Name);
+Console.WriteLine(updatedSecret.Value);
+Console.WriteLine(updatedSecret.Version);
+Console.WriteLine(updatedSecret.ContentType);
+```
+
+### Delete a Secret
+`Delete` deletes a secret previously stored in the Key Vault. When [soft-delete][soft_delete] is not enabled for the Key Vault, this operation permanently deletes the secret.
+
+```c#
+DeletedSecret secret = client.Delete("secret-name");
+
+Console.WriteLine(secret.Name);
+Console.WriteLine(secret.Value);
+```
+
+### List secrets
+This example lists all the secrets in the specified Key Vault.
+
+```c#
+List<Response<SecretBase>> allSecrets = Client.GetSecrets();
+
+  foreach (Secret secret in allSecrets)
+  {
+    Console.WriteLine(secret.Name);
+  }
+```
+
+### Async operations
```

Comment:
Considering the different type of this section do you think it would make sense to make it ## Async Examples and remove from the list in Sync (then renamed to Sync) to have higher visibility for this?

Reply:
Good point. Modified